auth/google.go

```diff
@@ -15,49 +15,99 @@
 package auth
 
 import (
+	"encoding/json"
 	"fmt"
+	"log"
 	"net/http"
+	"os"
 	"os/user"
 	"path/filepath"
 
-	"github.com/coreos/mantle/Godeps/_workspace/src/code.google.com/p/goauth2/oauth"
+	"github.com/coreos/mantle/Godeps/_workspace/src/golang.org/x/oauth2"
 )
 
 // client registered under 'marineam-tools'
-var config = oauth.Config{
-	ClientId:     "937427706989-nbndmfkp0knqardoagk6lbcamrsh828i.apps.googleusercontent.com",
+var conf = oauth2.Config{
+	ClientID:     "937427706989-nbndmfkp0knqardoagk6lbcamrsh828i.apps.googleusercontent.com",
 	ClientSecret: "F6Xs5wGHZzGw-QFXl3aylLUT",
-	Scope:        "https://www.googleapis.com/auth/devstorage.full_control",
-	AuthURL:      "https://accounts.google.com/o/oauth2/auth",
-	TokenURL:     "https://accounts.google.com/o/oauth2/token",
-	RedirectURL:  "urn:ietf:wg:oauth:2.0:oob",
+	Endpoint: oauth2.Endpoint{
+		AuthURL:  "https://accounts.google.com/o/oauth2/auth",
+		TokenURL: "https://accounts.google.com/o/oauth2/token",
+	},
+	RedirectURL: "urn:ietf:wg:oauth:2.0:oob",
+	Scopes:      []string{"https://www.googleapis.com/auth/devstorage.full_control"},
 }
 
-func GoogleClient(debug bool) (*http.Client, error) {
+func writeCache(cachePath string, tok *oauth2.Token) error {
+	file, err := os.OpenFile(cachePath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0600)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	if err := json.NewEncoder(file).Encode(tok); err != nil {
+		return err
+	}
+	return nil
+}
+
+func readCache(cachePath string) (*oauth2.Token, error) {
+	file, err := os.Open(cachePath)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	tok := &oauth2.Token{}
+	if err := json.NewDecoder(file).Decode(tok); err != nil {
+		return nil, err
+	}
+	return tok, nil
+}
+
+func getToken() (*oauth2.Token, error) {
 	userInfo, err := user.Current()
 	if err != nil {
 		return nil, err
 	}
 
-	cachePath := filepath.Join(userInfo.HomeDir, ".gsextra-google.json")
-	transport := oauth.Transport{Config: &config}
-	transport.TokenCache = oauth.CacheFile(cachePath)
-	if debug {
-		transport.Transport = &logTransport{http.DefaultTransport}
+	cachePath := filepath.Join(userInfo.HomeDir, ".tokencache-google.json")
```

Comment:
`.mantle-google.json` is probably better so someone has a better chance of figuring out where that file came from.